open-metadata-resources/open-metadata-samples/open-metadata-security-samples/src/main/java/org/odpi/openmetadata/metadatasecurity/samples/CocoPharmaServerLDAPSecurityConnector.java

```diff
@@ -0,0 +1,195 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+package org.odpi.openmetadata.metadatasecurity.samples;
+
+import java.util.Properties;
+import java.util.StringTokenizer;
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.*;
+
+public class CocoPharmaServerLDAPSecurityConnector extends AbstractCocoPharmaServerSecurityConnector {
+    private DirContext connection;
+    private String[] firstNames = {"Zach", "Steve", "Terri", "Tanya", "Polly", "Tessa", "Callie",
+            "Ivor", "Bob", "Faith", "Sally", "Lemmie", "Erin", "Harry", " Gary", "Grant", "Robbie", "Reggie",
+            "Peter", "Nancy", "Sidney", "Tom", "Julie", "Des", "Angela", "Jules", "Stew"};
+    private String[] lastNames = {"Now", "Starter", "Daring", "Tidie", "Tasker", "Tube", "Quartile",
+            "Padlock", "Nitter", "Broker", "Counter", "Stage", "Overview", "Hopeful", "Geeke", "Able", "Records",
+            "Mint", "Profile", "Noah", "Seeker", "Tally", "Stiched", "Signa", "Cumming", "Keeper", "Faster"};
+
+    public void makeAConnection() throws NamingException {
+        //Using Java Properties to create a connection to the LDAP server present in Apache LDAP Directory
+        Properties environment = new Properties();
+        environment.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+        environment.put(Context.PROVIDER_URL, "ldap://localhost:10389"); //Server details
+        environment.put(Context.SECURITY_PRINCIPAL, "uid=admin, ou=system"); //Current Admin User
+        environment.put(Context.SECURITY_CREDENTIALS, "secret"); //Admin Password
+
+        this.connection = new InitialDirContext(environment);
+        //Change this to Java Logging later
+        System.out.println("Trying connection to LDAP Apache Server....\n" + connection);
+    }
+
+    /**
+     * Searches currently connected LDAP Apache Server for all users and prints them to console.
+     * @throws NamingException
+     */
+    public void getAllUsers() throws NamingException {
+        String searchFilter = "(objectClass=inetOrgPerson)";
+        String[] regAttribute = {"cn"}; //Searching by Common Name, could add more filters if necessary
+        SearchControls controls = new SearchControls();
+        controls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+        controls.setReturningAttributes(regAttribute);
+
+        NamingEnumeration users = connection.search("ou=users,ou=system" ,searchFilter ,controls);
+
+        SearchResult result = null;
+        while (users.hasMore()) {
+            result = (SearchResult) users.next();
+            Attributes attribute = result.getAttributes();
+            System.out.println(attribute.get("cn"));
+        }
+    }
+
+    /**
+     * Adds an individual user to the LDAP Apache Server
```

Comment:
> Generally we will only have READ access to an ldap server. in a demo environment we have flexibility to write, and it may be helpful as part of the development of the connector, but I think it would be more typical to see .ldif files provided for the required users, and in future groups (we have some in Egeria, but they are incomplete). So I don't think we'd need to do any writes/adds within the connector

Yes I agree, we should not be writing user of group content to LDAP in this connector.

Reply:
Thank you for this! 
I was experimenting with what was and wasn't possible with the LDAP server and will make the change to using LDIF files and removing the possibility to write to the server. 

Noted!